liquibase-standard/src/main/java/liquibase/datatype/core/NVarcharType.java

```diff
@@ -28,24 +31,48 @@
+            if (originalDefinition.toLowerCase(Locale.US).startsWith("ntext") ||
+                    originalDefinition.toLowerCase(Locale.US).startsWith("[ntext]")){
+                if (! Boolean.TRUE.equals(GlobalConfiguration.CONVERT_DATA_TYPES.getCurrentValue())) {
+                    return new DatabaseDataType(database.escapeDataTypeName(originalDefinition));
                 }
+                DatabaseDataType type = new DatabaseDataType(database.escapeDataTypeName("nvarchar"));
+                // If there is additional specification after ntext (e.g.  COLLATE), import that.
+                String originalExtraInfo = originalDefinition.replaceFirst("^(?i)\\[?ntext\\]?\\s*", "");
+                originalExtraInfo = originalExtraInfo.replaceFirst("^(?i)\\[?text\\]?\\s*", "");
+                type.addAdditionalInformation("(max)");
+                if(!StringUtil.isEmpty(originalExtraInfo)) {
+                    //if we still have something like (25555) remove it
+                    //since we already set it to max, otherwise add collate or other info
+                    if(originalExtraInfo.lastIndexOf(")") < (originalExtraInfo.length() - 1)) {
+                        type.addAdditionalInformation(originalExtraInfo.substring(originalExtraInfo.lastIndexOf(")") + 1));
+                    }
+                }
+                return type;
+            } else {
+                Object[] parameters = getParameters();
+                if (parameters.length > 0) {
+                    String param1 = parameters[0].toString();
+                    if (!param1.matches("\\d+") || (new BigInteger(param1).compareTo(BigInteger.valueOf(4000L)) > 0)) {
+
+                        DatabaseDataType type = new DatabaseDataType(database.escapeDataTypeName("nvarchar"), "MAX");
+                        type.addAdditionalInformation(getAdditionalInformation());
+                        return type;
+                    }
+                }
+                if (parameters.length == 0) {
+                    parameters = new Object[] { 1 };
+                } else if (parameters.length > 1) {
+                    parameters = Arrays.copyOfRange(parameters, 0, 1);
+                }
+                DatabaseDataType type =  new DatabaseDataType(database.escapeDataTypeName("nvarchar"), parameters);
+                type.addAdditionalInformation(getAdditionalInformation());
+                return type;
             }
-            if (parameters.length == 0) {
-                parameters = new Object[] { 1 };
-            } else if (parameters.length > 1) {
-                parameters = Arrays.copyOfRange(parameters, 0, 1);
-            }
-            DatabaseDataType type =  new DatabaseDataType(database.escapeDataTypeName("nvarchar"), parameters);
-            type.addAdditionalInformation(getAdditionalInformation());
-            return type;
+        } else if ((database instanceof PostgresDatabase) || (database instanceof SQLiteDatabase) || (database
```

Comment:
Yeah, this one is contradictory because in L25 we have this:

```java
  if ((database instanceof HsqlDatabase) || (database instanceof PostgresDatabase) || (database instanceof
            DerbyDatabase)) {
```

and then on L73 inside an `else` block we have  `if ((database instanceof PostgresDatabase)`.

Could you please take a look at it?